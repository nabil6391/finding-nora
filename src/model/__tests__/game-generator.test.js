import { generateGridWithPuzzle } from '../game-generator'

const SOLUTION = (name || 'NORA').toUpperCase().split('')
const NOISE = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'.split('')

it('Generated grid should match snapshot', () => {
  const grid = generateGridWithPuzzle({
    noise: NOISE.filter(l => l !== SOLUTION[0]), // remove first letter of solution from noise array to improve gameplay
    size: 5,
    solution: SOLUTION,
  })
  expect(grid).toMatchSnapshot()
})
