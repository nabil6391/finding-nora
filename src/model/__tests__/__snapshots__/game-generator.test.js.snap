// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generated grid should match snapshot 1`] = `
Array [
  Array [
    Object {
      "column": 0,
      "letter": "X",
      "row": 0,
      "updatedAt": "2017-07-14T02:40:00.000Z",
    },
    Object {
      "column": 1,
      "letter": "H",
      "row": 0,
      "updatedAt": "2017-07-14T02:40:00.000Z",
    },
    Object {
      "column": 2,
      "letter": "M",
      "row": 0,
      "updatedAt": "2017-07-14T02:40:00.000Z",
    },
    Object {
      "column": 3,
      "letter": "V",
      "row": 0,
      "updatedAt": "2017-07-14T02:40:00.000Z",
    },
    Object {
      "column": 4,
      "letter": "S",
      "row": 0,
      "updatedAt": 2017-07-14T02:40:00.000Z,
    },
  ],
  Array [
    Object {
      "column": 0,
      "letter": "A",
      "row": 1,
      "updatedAt": "2017-07-14T02:40:00.000Z",
    },
    Object {
      "column": 1,
      "letter": "O",
      "row": 1,
      "updatedAt": "2017-07-14T02:40:00.000Z",
    },
    Object {
      "column": 2,
      "letter": "D",
      "row": 1,
      "updatedAt": "2017-07-14T02:40:00.000Z",
    },
    Object {
      "column": 3,
      "letter": "E",
      "row": 1,
      "updatedAt": "2017-07-14T02:40:00.000Z",
    },
    Object {
      "column": 4,
      "letter": "J",
      "row": 1,
      "updatedAt": "2017-07-14T02:40:00.000Z",
    },
  ],
  Array [
    Object {
      "column": 0,
      "letter": "A",
      "row": 2,
      "updatedAt": "2017-07-14T02:40:00.000Z",
    },
    Object {
      "column": 1,
      "letter": "N",
      "row": 2,
      "updatedAt": "2017-07-14T02:40:00.000Z",
    },
    Object {
      "column": 2,
      "letter": "C",
      "row": 2,
      "updatedAt": "2017-07-14T02:40:00.000Z",
    },
    Object {
      "column": 3,
      "letter": "J",
      "row": 2,
      "updatedAt": "2017-07-14T02:40:00.000Z",
    },
    Object {
      "column": 4,
      "letter": "G",
      "row": 2,
      "updatedAt": "2017-07-14T02:40:00.000Z",
    },
  ],
  Array [
    Object {
      "column": 0,
      "letter": "G",
      "row": 3,
      "updatedAt": "2017-07-14T02:40:00.000Z",
    },
    Object {
      "column": 1,
      "letter": "L",
      "row": 3,
      "updatedAt": "2017-07-14T02:40:00.000Z",
    },
    Object {
      "column": 2,
      "letter": "G",
      "row": 3,
      "updatedAt": "2017-07-14T02:40:00.000Z",
    },
    Object {
      "column": 3,
      "letter": "T",
      "row": 3,
      "updatedAt": "2017-07-14T02:40:00.000Z",
    },
    Object {
      "column": 4,
      "letter": "O",
      "row": 3,
      "updatedAt": "2017-07-14T02:40:00.000Z",
    },
  ],
  Array [
    Object {
      "column": 0,
      "letter": "U",
      "row": 4,
      "updatedAt": "2017-07-14T02:40:00.000Z",
    },
    Object {
      "column": 1,
      "letter": "T",
      "row": 4,
      "updatedAt": "2017-07-14T02:40:00.000Z",
    },
    Object {
      "column": 2,
      "letter": "G",
      "row": 4,
      "updatedAt": "2017-07-14T02:40:00.000Z",
    },
    Object {
      "column": 3,
      "letter": "R",
      "row": 4,
      "updatedAt": "2017-07-14T02:40:00.000Z",
    },
    Object {
      "column": 4,
      "letter": "E",
      "row": 4,
      "updatedAt": "2017-07-14T02:40:00.000Z",
    },
  ],
]
`;
